adobe/react-spectrum
# Calendar calendar state context set value can not set null value

Closes #6006

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/6006).

## 📝 Test Instructions:

1. Open the calendar page where the Footer component is inserted ([Story link](http://localhost:9003/?path=/story/react-aria-components--calendar-reset-value&providerSwitcher-express=false&strict=true))
2. Select an arbitrary date in the calendar.
3. Click the "Reset value" button in the Footer component.
4. Make sure that after clicking the button, the selected date is reset, and no date is displayed in the calendar as selected.
5. Check that pressing the "Reset value" button again after selecting a new date resets the selection again.


https://github.com/adobe/react-spectrum/assets/1297755/8b9f63cc-e333-41d8-81eb-119938665473
